packages/client/src/connectors/dynamic.ts

```diff
@@ -0,0 +1,229 @@
+
+import {
+  type Address,
+  MethodNotSupportedRpcError,
+  UserRejectedRequestError,
+} from 'viem'
+import { type Connection, ProviderNotFoundError, createConnector } from 'wagmi'
+import type { UTXOConnectorParameters } from './types.js'
+
+export type DynamicWalletConnectorEventMap = {
+  accountChange(props: { accounts: string[] }): void
+}
+
+export type DynamicWalletConnectorEvents = {
+  addListener<TEvent extends keyof DynamicWalletConnectorEventMap>(
+    event: TEvent,
+    listener: DynamicWalletConnectorEventMap[TEvent]
+  ): void
+  removeListener<TEvent extends keyof DynamicWalletConnectorEventMap>(
+    event: TEvent,
+    listener: DynamicWalletConnectorEventMap[TEvent]
+  ): void
+}
+
+export type DynamicWalletConnector = {
+  providerId: string
+  name: string
+  id: string
+
+  getAddress(): string
+  _metadata: {
+    icon?: string
+  }
+} & DynamicWalletConnectorEvents
+
+type BitcoinAddress = {
+  address: string
+  type: 'ordinals' | 'payment'
+  publicKey: string
+}
+
+export type BitcoinSignPsbtRequestSignature = {
+  address: string
+  signingIndexes: number[] | undefined
+  disableAddressValidation?: boolean
+}
+
+type BitcoinSignPsbtRequest = {
+  allowedSighash: number[]
+  unsignedPsbtBase64: string
+  signature?: BitcoinSignPsbtRequestSignature[]
+}
+
+type BitcoinSignPsbtResponse = {
+  signedPsbt: string
+}
+
+type DynamicBitcoinWallet = {
+  connector: DynamicWalletConnector
+  additionalAddresses: BitcoinAddress[]
+  address: string
+  isAuthenticated: boolean
+  signPsbt(
+    parameters: BitcoinSignPsbtRequest
+  ): Promise<BitcoinSignPsbtResponse | undefined>
+}
+
+type DynamicConnectorProperties = {
+  getAccounts(): Promise<readonly Address[]>
+  onAccountsChanged(accounts: Address[]): void
+  getInternalProvider(): Promise<DynamicWalletConnector>
+} & UTXOWalletProvider
+
+type DynamicConnectorParameters = {
+  wallet: DynamicBitcoinWallet
+} & UTXOConnectorParameters
+
+dynamic.type = 'UTXO' as const
+
+export function dynamic(parameters: DynamicConnectorParameters) {
+  const { chainId, shimDisconnect = true, wallet } = parameters
+  let accountChanged: ((accounts: string[]) => void) | undefined
+  return createConnector<
+    UTXOWalletProvider | undefined,
+    DynamicConnectorProperties
+  >((config) => ({
+    id: wallet.connector.providerId,
+    name: wallet.connector.name,
+    type: dynamic.type,
+    icon: wallet.connector._metadata?.icon,
+    emitter: config.emitter,
+
+    async isAuthorized() {
+      return wallet.isAuthenticated
+    },
+    async request(
+      this: DynamicBitcoinWallet,
+      { method, params }
+    ): Promise<any> {
+      switch (method) {
+        case 'signPsbt': {
+          const { psbt, ...options } = params as SignPsbtParameters
+          const allowedSighash: number[] = options.inputsToSign.map((input) =>
+            Number(input.sigHash)
+          )
+          const psbtBase64 = Buffer.from(psbt, 'hex').toString('base64')
```

Comment:
Buffer requires a polyfill, please take a look at other connectors on how to convert psbt without the buffer.